create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -1,52 +1,117 @@
 package no.sikt.nva.scopus;
 
+import static no.sikt.nva.scopus.ScopusConstants.ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME;
+import static nva.commons.core.attempt.Try.attempt;
 import com.amazonaws.services.lambda.runtime.Context;
 import com.amazonaws.services.lambda.runtime.RequestHandler;
 import com.amazonaws.services.lambda.runtime.events.S3Event;
 import jakarta.xml.bind.JAXB;
+import java.io.StringReader;
+import java.net.URI;
+import java.time.Instant;
 import no.scopus.generated.DocTp;
+import no.unit.nva.events.models.EventReference;
 import no.unit.nva.metadata.CreatePublicationRequest;
+import no.unit.nva.model.AdditionalIdentifier;
 import no.unit.nva.s3.S3Driver;
+import nva.commons.core.Environment;
 import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UnixPath;
 import nva.commons.core.paths.UriWrapper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.eventbridge.EventBridgeClient;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
 import software.amazon.awssdk.services.s3.S3Client;
 
-import java.io.StringReader;
-import java.net.URI;
-
-import static nva.commons.core.attempt.Try.attempt;
-
 public class ScopusHandler implements RequestHandler<S3Event, CreatePublicationRequest> {
 
     public static final int SINGLE_EXPECTED_RECORD = 0;
     public static final String S3_URI_TEMPLATE = "s3://%s/%s";
     private static final Logger logger = LoggerFactory.getLogger(ScopusHandler.class);
+    private static final String SCOPUS_EVENTS_FOLDER = "scopus/create";
+    //TODO: move to config class.
+    private static final String EVENTS_BUCKET = new Environment().readEnv("EVENTS_BUCKET");
+    private static final String EVENT_BUS = new Environment().readEnv("EVENTS_BUS");
+    public static final String EVENT_TOPIC_IS_SET_IN_EVENT_BODY = "ReferToEventTopic";
+
     private final S3Client s3Client;
+    private final EventBridgeClient eventBridgeClient;
 
     @JacocoGenerated
     public ScopusHandler() {
-        this(S3Driver.defaultS3Client().build());
+        this(S3Driver.defaultS3Client().build(), defaultEventBridgeClient());
     }
 
-    public ScopusHandler(S3Client s3Client) {
+    public ScopusHandler(S3Client s3Client, EventBridgeClient eventBridgeClient) {
         this.s3Client = s3Client;
+        this.eventBridgeClient = eventBridgeClient;
     }
 
     @Override
     public CreatePublicationRequest handleRequest(S3Event event, Context context) {
+        var request = createPublicationRequest(event);
+        emitEventToEventBridge(context, request);
```

Comment:
Hva jeg lurer på er hvorfor vi egentlig bruker ikke sammen måte som create-publication-from-doi? Hva det ikke grunnen til at vi holder på i dette repository i utgangspunktet. Siden vi gjør neste det samme "importere publicationer" ...
I create-publication-from-doi bruker vi PublicationPersistenceService.createPublication().
Faktisk liker jeg dene event driven approach bedre. Men det er litt inkonsistent.

Reply:
When a user visits the website and chooses to import a publication from doi we have their credentials, that we can use to create new publication. This is an automated process, and we should either use events or use an internal api. 

I went for the internal api in the beginning, but then there was a question what happens, when the Creation of a Publication fails. After a discussion with Rurik, we concluded that it not the responsibility of FetchDoi service to handle such events but the responsibility of the PublicationService. At that point it became apparent that FetchDoi should only say "here is a new entry from scopus , do whatever you want" *

*Note: Ideally the conversion code should be in the PublicationService and the FetchDoi should only send the scopus document. But this is another topic.